src/core/print.ml

```diff
@@ -36,24 +36,27 @@ let pp_assoc : assoc pp = fun oc assoc ->
   Format.fprintf oc
     (match assoc with
      | Assoc_none -> ""
-     | Assoc_left -> "left"
-     | Assoc_right -> "right")
+     | Assoc_left -> " left"
+     | Assoc_right -> " right")
 
 (** [hint oc a] prints hint [h] to channel [oc]. *)
 let pp_hint : pp_hint pp = fun oc pp_hint ->
   match pp_hint with
-  | No_hint -> ()
+  | Unqual -> ()
   | Prefix(n,p,_) -> Format.fprintf oc "prefix %s %f" n p
-  | Infix(n,a,p,_) -> Format.fprintf oc "infix %s %a %f" n pp_assoc a p
+  | Infix(n,a,p,_) -> Format.fprintf oc "infix %s%a %f" n pp_assoc a p
```

Comment:
And we might need to set some convention to quote terms of the language, we used `[f]` for some time but I understood you preferred `"f"`, we can use any of the two, but it would be good to remain consistent and always mark the elements that are from the language while printing.

Reply:
About pp_hint printing: it's mainly used for debugging but, yes, we can have more verbose messages.

About uniformity: you are right, it would be nice to be more uniform. But I also want to have something light. [] may be useful when to clearly distinguish multiple terms in a message. But, for a function symbol, I do not see the need, unless there may be some confusion with other elements of the message. I use "" for builtins as this is currently the syntax used to input them. If you spot places that could be fixed, let me know.